akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala

```diff
@@ -536,6 +536,19 @@ object TestSubscriber {
       }
     }
 
+    /**
+     * Expect next element and test it with partial function.
+     *
+     * Allows chaining probe methods.
+     */
+    def expectNextChainingPF(f: PartialFunction[Any, Any]): Self = {
+      expectEventPF {
+        case OnNext(n) ⇒
+          assert(f.isDefinedAt(n))
```

Comment:
improved code and messages, now it will return: "expected: message matching partial function but got unexpected message $n"